syntax = "proto3";

service serverService{
	rpc trainerCheck(your_name) returns(Spaces);
	rpc moveTrainer(Decision) returns(Valid);
	rpc movePokemon(Decision) returns(Valid);
	rpc pokemonCheck(your_name) returns(PSpaces);
}

message your_name{
	string name = 1;
}

message Decision{
	repeated int32 move2 = 1;
	string name = 2; 
	repeated int32 cur_pos1 = 3;
	int32 capture = 4;
}


message Spaces{
	repeated int32 pokemons_near = 1;
	repeated int32 emt_spaces = 2 ;
	int32 n_pokemons = 3;
	repeated int32 cur_pos = 4;
	int32 loc = 5;
}

message PSpaces{
	repeated int32 valid_mov = 1;
	repeated int32 valid_mov_before = 2 ;
	repeated int32 cur_posp = 3;
	int32 loc1 = 4;
}
message Valid{
	string move_status = 1;
}
